server/channels/store/sqlstore/store.go

```diff
@@ -1002,18 +1003,57 @@ func (ss *SqlStore) DropAllTables() {
 			    FROM   pg_class
 			    WHERE  relkind = 'r'  -- only tables
 			    AND    relnamespace = 'public'::regnamespace
-				AND NOT relname = 'db_migrations'
+				AND NOT (
+                  relname = 'db_migrations' OR
+                  relname = 'focalboard_schema_migrations' OR
+                  relname = 'focalboard_boards' OR
+                  relname = 'focalboard_blocks'
+                )
 			   );
 			END
 			$func$;`)
+		tableSchemaFn = "current_schema()"
 	} else {
 		tables := []string{}
 		ss.masterX.Select(&tables, `show tables`)
 		for _, t := range tables {
-			if t != "db_migrations" {
+			if t != "db_migrations" &&
+				t != "focalboard_schema_migrations" &&
+				t != "focalboard_boards" &&
+				t != "focalboard_blocks" {
 				ss.masterX.Exec(`TRUNCATE TABLE ` + t)
+
 			}
 		}
+		tableSchemaFn = "DATABASE()"
+	}
+
+	var boardsTableCount int
+	err := ss.masterX.Get(&boardsTableCount, `
+      SELECT COUNT(*)
+        FROM INFORMATION_SCHEMA.TABLES
+       WHERE TABLE_SCHEMA = `+tableSchemaFn+`
+         AND TABLE_NAME = 'focalboard_schema_migrations'`)
+	if err != nil {
+		panic(errors.Wrap(err, "Error dropping all tables. Cannot query INFORMATION_SCHEMA table to check for focalboard_schema_migrations table"))
+	}
+
+	if boardsTableCount != 0 {
+		_, blErr := ss.masterX.Exec(`
+          DELETE FROM focalboard_blocks
```

Comment:
Where are the non-template boards being deleted?

Reply:
In the next query: https://github.com/mattermost/mattermost-server/pull/22757/files/e02d8d149c0d128172c76fc2397c8f854770c84f#diff-98025905572bcd0aededdab8fe812f6bb3bda62b2777b1f8e5cd9f23aeebdfccR1053

Reply:
Sorry - missed that.